Index: system/ScriptLoader.cpp
===================================================================
--- system/ScriptLoader.cpp	(révision 1310)
+++ system/ScriptLoader.cpp	(copie de travail)
@@ -5,6 +5,12 @@
 #include "precompiled.h"
 
 //custom
+// -- MISS --
+extern void AddSC_mercenary_summoner();
+extern void AddSC_mercenary_shaman();
+extern void AddSC_mercenary_warlock();
+extern void AddSC_mercenary_paladin();
+extern void AddSC_mercenary_mage();
 
 //examples
 extern void AddSC_example_creature();
@@ -356,7 +362,12 @@
 void AddScripts()
 {
     //custom
-
+    // -- MISS --
+		AddSC_mercenary_summoner();
+		AddSC_mercenary_shaman();
+		AddSC_mercenary_warlock();
+		AddSC_mercenary_paladin();
+		AddSC_mercenary_mage();
     //examples
     AddSC_example_creature();
     AddSC_example_escort();
Index: VC90/90ScriptDev2.vcproj
===================================================================
--- VC90/90ScriptDev2.vcproj	(révision 1310)
+++ VC90/90ScriptDev2.vcproj	(copie de travail)
@@ -391,10 +391,154 @@
 			<Filter
 				Name="custom"
 				>
+				<Filter
+					Name="miss"
+					>
+					<File
+						RelativePath="..\scripts\MISS\Shared\defines.cpp"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\Shared\defines.h"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\mage.cpp"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\mage.h"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\paladin.cpp"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\paladin.h"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\shaman.cpp"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\shaman.h"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\summoner.cpp"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\warlock.cpp"
+						>
+					</File>
+					<File
+						RelativePath="..\scripts\MISS\AI\warlock.h"
+						>
+					</File>
+				</Filter>
 			</Filter>
 			<Filter
 				Name="eastern_kingdoms"
 				>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\alterac_mountains.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\arathi_highlands.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\blasted_lands.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\boss_kruul.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\burning_steppes.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\dun_morogh.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\eastern_plaguelands.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\elwynn_forest.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\eversong_woods.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\ghostlands.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\hinterlands.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\ironforge.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\isle_of_queldanas.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\loch_modan.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\searing_gorge.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\silvermoon_city.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\silverpine_forest.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\stormwind_city.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\stranglethorn_vale.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\tirisfal_glades.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\undercity.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\western_plaguelands.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\westfall.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\eastern_kingdoms\wetlands.cpp"
+					>
+				</File>
 				<Filter
 					Name="blackrock_depths"
 					>
@@ -1043,126 +1187,114 @@
 						>
 					</File>
 				</Filter>
+			</Filter>
+			<Filter
+				Name="examples"
+				>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\alterac_mountains.cpp"
+					RelativePath="..\scripts\examples\example_creature.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\arathi_highlands.cpp"
+					RelativePath="..\scripts\examples\example_escort.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\blasted_lands.cpp"
+					RelativePath="..\scripts\examples\example_gossip_codebox.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\boss_kruul.cpp"
+					RelativePath="..\scripts\examples\example_misc.cpp"
 					>
 				</File>
+			</Filter>
+			<Filter
+				Name="kalimdor"
+				>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\burning_steppes.cpp"
+					RelativePath="..\scripts\kalimdor\ashenvale.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\dun_morogh.cpp"
+					RelativePath="..\scripts\kalimdor\azshara.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\eastern_plaguelands.cpp"
+					RelativePath="..\scripts\kalimdor\azuremyst_isle.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\elwynn_forest.cpp"
+					RelativePath="..\scripts\kalimdor\bloodmyst_isle.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\eversong_woods.cpp"
+					RelativePath="..\scripts\kalimdor\boss_azuregos.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\ghostlands.cpp"
+					RelativePath="..\scripts\kalimdor\darkshore.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\hinterlands.cpp"
+					RelativePath="..\scripts\kalimdor\desolace.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\ironforge.cpp"
+					RelativePath="..\scripts\kalimdor\dustwallow_marsh.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\isle_of_queldanas.cpp"
+					RelativePath="..\scripts\kalimdor\felwood.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\loch_modan.cpp"
+					RelativePath="..\scripts\kalimdor\feralas.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\searing_gorge.cpp"
+					RelativePath="..\scripts\kalimdor\moonglade.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\silvermoon_city.cpp"
+					RelativePath="..\scripts\kalimdor\mulgore.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\silverpine_forest.cpp"
+					RelativePath="..\scripts\kalimdor\orgrimmar.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\stormwind_city.cpp"
+					RelativePath="..\scripts\kalimdor\silithus.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\stranglethorn_vale.cpp"
+					RelativePath="..\scripts\kalimdor\stonetalon_mountains.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\tirisfal_glades.cpp"
+					RelativePath="..\scripts\kalimdor\tanaris.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\undercity.cpp"
+					RelativePath="..\scripts\kalimdor\teldrassil.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\western_plaguelands.cpp"
+					RelativePath="..\scripts\kalimdor\the_barrens.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\westfall.cpp"
+					RelativePath="..\scripts\kalimdor\thousand_needles.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\eastern_kingdoms\wetlands.cpp"
+					RelativePath="..\scripts\kalimdor\thunder_bluff.cpp"
 					>
 				</File>
-			</Filter>
-			<Filter
-				Name="examples"
-				>
 				<File
-					RelativePath="..\scripts\examples\example_creature.cpp"
+					RelativePath="..\scripts\kalimdor\winterspring.cpp"
 					>
 				</File>
-				<File
-					RelativePath="..\scripts\examples\example_escort.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\examples\example_gossip_codebox.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\examples\example_misc.cpp"
-					>
-				</File>
-			</Filter>
-			<Filter
-				Name="kalimdor"
-				>
 				<Filter
 					Name="blackfathom_deeps"
 					>
@@ -1419,94 +1551,26 @@
 						>
 					</File>
 				</Filter>
+			</Filter>
+			<Filter
+				Name="northrend"
+				>
 				<File
-					RelativePath="..\scripts\kalimdor\ashenvale.cpp"
+					RelativePath="..\scripts\northrend\borean_tundra.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\kalimdor\azshara.cpp"
+					RelativePath="..\scripts\northrend\dragonblight.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\kalimdor\azuremyst_isle.cpp"
+					RelativePath="..\scripts\northrend\grizzly_hills.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\kalimdor\bloodmyst_isle.cpp"
+					RelativePath="..\scripts\northrend\icecrown.cpp"
 					>
 				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\boss_azuregos.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\darkshore.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\desolace.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\dustwallow_marsh.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\felwood.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\feralas.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\moonglade.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\mulgore.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\orgrimmar.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\silithus.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\stonetalon_mountains.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\tanaris.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\teldrassil.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\the_barrens.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\thousand_needles.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\thunder_bluff.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\kalimdor\winterspring.cpp"
-					>
-				</File>
-			</Filter>
-			<Filter
-				Name="northrend"
-				>
 				<Filter
 					Name="azjol-nerub"
 					>
@@ -1739,26 +1803,50 @@
 					Name="zuldrak"
 					>
 				</Filter>
+			</Filter>
+			<Filter
+				Name="outland"
+				>
 				<File
-					RelativePath="..\scripts\northrend\borean_tundra.cpp"
+					RelativePath="..\scripts\outland\blades_edge_mountains.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\northrend\dragonblight.cpp"
+					RelativePath="..\scripts\outland\boss_doomlord_kazzak.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\northrend\grizzly_hills.cpp"
+					RelativePath="..\scripts\outland\boss_doomwalker.cpp"
 					>
 				</File>
 				<File
-					RelativePath="..\scripts\northrend\icecrown.cpp"
+					RelativePath="..\scripts\outland\hellfire_peninsula.cpp"
 					>
 				</File>
-			</Filter>
-			<Filter
-				Name="outland"
-				>
+				<File
+					RelativePath="..\scripts\outland\nagrand.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\outland\netherstorm.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\outland\shadowmoon_valley.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\outland\shattrath_city.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\outland\terokkar_forest.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\outland\zangarmarsh.cpp"
+					>
+				</File>
 				<Filter
 					Name="aunchindoun"
 					>
@@ -2058,7 +2146,8 @@
 							>
 						</File>
 						<File
-							RelativePath="..\scripts\outland\hellfire_citadel\shattered_halls\boss_warchief_kargath_bladefist.cpp">
+							RelativePath="..\scripts\outland\hellfire_citadel\shattered_halls\boss_warchief_kargath_bladefist.cpp"
+							>
 						</File>
 						<File
 							RelativePath="..\scripts\outland\hellfire_citadel\shattered_halls\def_shattered_halls.h"
@@ -2166,46 +2255,6 @@
 						</File>
 					</Filter>
 				</Filter>
-				<File
-					RelativePath="..\scripts\outland\blades_edge_mountains.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\boss_doomlord_kazzak.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\boss_doomwalker.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\hellfire_peninsula.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\nagrand.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\netherstorm.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\shadowmoon_valley.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\shattrath_city.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\terokkar_forest.cpp"
-					>
-				</File>
-				<File
-					RelativePath="..\scripts\outland\zangarmarsh.cpp"
-					>
-				</File>
 			</Filter>
 			<Filter
 				Name="world"

