Index: ScriptMgr.cpp
===================================================================
--- ScriptMgr.cpp	(révision 1110)
+++ ScriptMgr.cpp	(copie de travail)
@@ -50,6 +50,13 @@
 // -- Creature --
 extern void AddSC_generic_creature();
 
+// -- MISS --
+extern void AddSC_mercenary_summoner();
+extern void AddSC_mercenary_shaman();
+extern void AddSC_mercenary_warlock();
+extern void AddSC_mercenary_paladin();
+extern void AddSC_mercenary_mage();
+
 // -- Custom --
 
 // -- Examples --
@@ -902,6 +909,13 @@
 
     // -- Creature --
     AddSC_generic_creature();
+    
+    // -- MISS --
+		AddSC_mercenary_summoner();
+		AddSC_mercenary_shaman();
+		AddSC_mercenary_warlock();
+		AddSC_mercenary_paladin();
+		AddSC_mercenary_mage();
 
     // -- Custom --
 
Index: VC90/90ScriptDev2.vcproj
===================================================================
--- VC90/90ScriptDev2.vcproj	(révision 1110)
+++ VC90/90ScriptDev2.vcproj	(copie de travail)
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="windows-1250"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="9.00"
+	Version="9,00"
 	Name="ScriptDev2"
 	ProjectGUID="{4295C8A9-79B7-4354-8064-F05FB9CA0C96}"
 	RootNamespace="ScriptDev2"
@@ -103,8 +103,8 @@
 			/>
 		</Configuration>
 		<Configuration
-			Name="Release|Win32"
-			OutputDirectory="..\..\..\..\bin\win32_release"
+			Name="Debug|x64"
+			OutputDirectory="..\..\..\..\bin\x64_debug"
 			IntermediateDirectory=".\ScriptDev2__$(PlatformName)_$(ConfigurationName)"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -125,14 +125,16 @@
 			/>
 			<Tool
 				Name="VCMIDLTool"
+				TargetEnvironment="3"
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalOptions="/MP"
+				Optimization="0"
 				AdditionalIncludeDirectories="..\..\..\..\dep\include\;..\..\..\shared\;..\..\..\framework\;..\..\..\game\;..\include\;..\..\..\..\dep\ACE_wrappers\"
-				PreprocessorDefinitions="WIN32;NDEBUG;_WINDOWS;_USRDLL;SCRIPT"
-				MinimalRebuild="false"
-				RuntimeLibrary="2"
+				PreprocessorDefinitions="WIN32;_DEBUG;MANGOS_DEBUG;_WINDOWS;_USRDLL;SCRIPT"
+				MinimalRebuild="true"
+				BasicRuntimeChecks="3"
+				RuntimeLibrary="3"
 				UsePrecompiledHeader="2"
 				PrecompiledHeaderThrough="precompiled.h"
 				WarningLevel="3"
@@ -150,18 +152,17 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="mangosd.lib ACE.lib"
+				AdditionalDependencies="mangosd.lib ACEd.lib"
 				OutputFile="$(OutDir)/MaNGOSScript.dll"
-				LinkIncremental="1"
-				AdditionalLibraryDirectories="..\..\..\..\win\VC90\mangosd__Win32_Release;..\..\..\..\dep\lib\win32_release\"
-				GenerateDebugInformation="false"
+				LinkIncremental="2"
+				AdditionalLibraryDirectories="..\..\..\..\win\VC90\mangosd__x64_Debug;..\..\..\..\dep\lib\x64_Debug\"
+				GenerateDebugInformation="true"
+				ProgramDatabaseFile="$(OutDir)/MaNGOSScript.pdb"
 				SubSystem="2"
-				OptimizeReferences="2"
-				EnableCOMDATFolding="2"
 				RandomizedBaseAddress="1"
 				DataExecutionPrevention="0"
 				ImportLibrary="$(OutDir)/MaNGOSScript.lib"
-				TargetMachine="1"
+				TargetMachine="17"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -186,8 +187,8 @@
 			/>
 		</Configuration>
 		<Configuration
-			Name="Debug|x64"
-			OutputDirectory="..\..\..\..\bin\x64_debug"
+			Name="Release|Win32"
+			OutputDirectory="..\..\..\..\bin\win32_release"
 			IntermediateDirectory=".\ScriptDev2__$(PlatformName)_$(ConfigurationName)"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -208,16 +209,14 @@
 			/>
 			<Tool
 				Name="VCMIDLTool"
-				TargetEnvironment="3"
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				Optimization="0"
+				AdditionalOptions="/MP"
 				AdditionalIncludeDirectories="..\..\..\..\dep\include\;..\..\..\shared\;..\..\..\framework\;..\..\..\game\;..\include\;..\..\..\..\dep\ACE_wrappers\"
-				PreprocessorDefinitions="WIN32;_DEBUG;MANGOS_DEBUG;_WINDOWS;_USRDLL;SCRIPT"
-				MinimalRebuild="true"
-				BasicRuntimeChecks="3"
-				RuntimeLibrary="3"
+				PreprocessorDefinitions="WIN32;NDEBUG;_WINDOWS;_USRDLL;SCRIPT"
+				MinimalRebuild="false"
+				RuntimeLibrary="2"
 				UsePrecompiledHeader="2"
 				PrecompiledHeaderThrough="precompiled.h"
 				WarningLevel="3"
@@ -235,17 +234,18 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="mangosd.lib ACEd.lib"
+				AdditionalDependencies="mangosd.lib ACE.lib"
 				OutputFile="$(OutDir)/MaNGOSScript.dll"
-				LinkIncremental="2"
-				AdditionalLibraryDirectories="..\..\..\..\win\VC90\mangosd__x64_Debug;..\..\..\..\dep\lib\x64_Debug\"
-				GenerateDebugInformation="true"
-				ProgramDatabaseFile="$(OutDir)/MaNGOSScript.pdb"
+				LinkIncremental="1"
+				AdditionalLibraryDirectories="..\..\..\..\win\VC90\mangosd__Win32_Release;..\..\..\..\dep\lib\win32_release\"
+				GenerateDebugInformation="false"
 				SubSystem="2"
+				OptimizeReferences="2"
+				EnableCOMDATFolding="2"
 				RandomizedBaseAddress="1"
 				DataExecutionPrevention="0"
 				ImportLibrary="$(OutDir)/MaNGOSScript.lib"
-				TargetMachine="17"
+				TargetMachine="1"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -1658,7 +1658,8 @@
 							>
 						</File>
 						<File
-							RelativePath="..\scripts\zone\hellfire_citadel\shattered_halls\boss_warchief_kargath_bladefist.cpp">
+							RelativePath="..\scripts\zone\hellfire_citadel\shattered_halls\boss_warchief_kargath_bladefist.cpp"
+							>
 						</File>
 						<File
 							RelativePath="..\scripts\zone\hellfire_citadel\shattered_halls\def_shattered_halls.h"
@@ -2459,6 +2460,54 @@
 					>
 				</Filter>
 			</Filter>
+			<Filter
+				Name="miss"
+				>
+				<File
+					RelativePath="..\scripts\MISS\Shared\defines.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\Shared\defines.h"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\mage.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\mage.h"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\paladin.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\paladin.h"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\shaman.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\shaman.h"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\summoner.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\warlock.cpp"
+					>
+				</File>
+				<File
+					RelativePath="..\scripts\MISS\AI\warlock.h"
+					>
+				</File>
+			</Filter>
 		</Filter>
 		<Filter
 			Name="Include"
@@ -2475,7 +2524,7 @@
 					/>
 				</FileConfiguration>
 				<FileConfiguration
-					Name="Release|Win32"
+					Name="Debug|x64"
 					>
 					<Tool
 						Name="VCCLCompilerTool"
@@ -2483,7 +2532,7 @@
 					/>
 				</FileConfiguration>
 				<FileConfiguration
-					Name="Debug|x64"
+					Name="Release|Win32"
 					>
 					<Tool
 						Name="VCCLCompilerTool"
@@ -2537,14 +2586,6 @@
 			>
 		</File>
 		<File
-			RelativePath="..\ScriptMgr.cpp"
-			>
-		</File>
-		<File
-			RelativePath="..\ScriptMgr.h"
-			>
-		</File>
-		<File
 			RelativePath="..\revision.h"
 			>
 			<FileConfiguration
@@ -2553,29 +2594,29 @@
 				<Tool
 					Name="VCCustomBuildTool"
 					Description="Extracting Revision"
-					CommandLine="echo on&#x0D;&#x0A;cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;..\..\..\win\VC90\genrevision__Win32_$(ConfigurationName)\genrevision.exe&quot;"
+					CommandLine="echo on&#x0D;&#x0A;cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;..\..\..\win\VC90\genrevision__Win32_$(ConfigurationName)\genrevision.exe&quot;&#x0D;&#x0A;"
 					AdditionalDependencies=".svn/entries"
 					Outputs="revision.h"
 				/>
 			</FileConfiguration>
 			<FileConfiguration
-				Name="Release|Win32"
+				Name="Debug|x64"
 				>
 				<Tool
 					Name="VCCustomBuildTool"
 					Description="Extracting Revision"
-					CommandLine="echo on&#x0D;&#x0A;cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;..\..\..\win\VC90\genrevision__Win32_$(ConfigurationName)\genrevision.exe&quot;"
+					CommandLine="echo on&#x0D;&#x0A;cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;..\..\..\win\VC90\genrevision__Win32_$(ConfigurationName)\genrevision.exe&quot;&#x0D;&#x0A;"
 					AdditionalDependencies=".svn/entries"
 					Outputs="revision.h"
 				/>
 			</FileConfiguration>
 			<FileConfiguration
-				Name="Debug|x64"
+				Name="Release|Win32"
 				>
 				<Tool
 					Name="VCCustomBuildTool"
 					Description="Extracting Revision"
-					CommandLine="echo on&#x0D;&#x0A;cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;..\..\..\win\VC90\genrevision__Win32_$(ConfigurationName)\genrevision.exe&quot;"
+					CommandLine="echo on&#x0D;&#x0A;cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;..\..\..\win\VC90\genrevision__Win32_$(ConfigurationName)\genrevision.exe&quot;&#x0D;&#x0A;"
 					AdditionalDependencies=".svn/entries"
 					Outputs="revision.h"
 				/>
@@ -2586,13 +2627,21 @@
 				<Tool
 					Name="VCCustomBuildTool"
 					Description="Extracting Revision"
-					CommandLine="echo on&#x0D;&#x0A;cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;..\..\..\win\VC90\genrevision__Win32_$(ConfigurationName)\genrevision.exe&quot;"
+					CommandLine="echo on&#x0D;&#x0A;cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;..\..\..\win\VC90\genrevision__Win32_$(ConfigurationName)\genrevision.exe&quot;&#x0D;&#x0A;"
 					AdditionalDependencies=".svn/entries"
 					Outputs="revision.h"
 				/>
 			</FileConfiguration>
 		</File>
 		<File
+			RelativePath="..\ScriptMgr.cpp"
+			>
+		</File>
+		<File
+			RelativePath="..\ScriptMgr.h"
+			>
+		</File>
+		<File
 			RelativePath="..\system.cpp"
 			>
 		</File>
